core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -693,6 +693,11 @@ private[spark] class DAGScheduler(
 
     val jobId = nextJobId.getAndIncrement()
     if (partitions.size == 0) {
+      val time = clock.getTimeMillis()
```

Comment:
This is looking OK to me, though I wouldn't mind, say, @cloud-fan taking a quick look.

Reply:
@srowen Thank you! @cloud-fan Could you give me some suggestions?